lib/kitchen/directions/bake_injected_exercise/bake_injected_exercise.rb

```diff
@@ -1,17 +1,71 @@
 # frozen_string_literal: true
 
 module Kitchen::Directions::BakeInjectedExercise
-  def self.v1(exercise:)
-    V1.new.bake(exercise: exercise)
+  def self.v1(exercise:, alphabetical_multiparts: false)
+    V1.new.bake(exercise: exercise, alphabetical_multiparts: alphabetical_multiparts)
   end
 
   class V1
-    def bake(exercise:)
+    def bake(exercise:, alphabetical_multiparts:)
       question_count = exercise.injected_questions.count
       exercise[:'data-question-count'] = question_count
       exercise[:'data-is-multipart'] = question_count > 1 ? 'True' : 'False'
 
       context = exercise&.exercise_context
+      stimulus = exercise&.first("div[data-type='exercise-stimulus']")
+
+      # To handle alphabetical multipart questions without interrupting numbering
+      if alphabetical_multiparts && stimulus
+        solutions_clipboard = Kitchen::Clipboard.new
+        questions_clipboard = Kitchen::Clipboard.new
+        alphabet = *('a'..'z')
+        wrapper_id = "#{exercise.injected_questions.first.id}-wrapper"
+
+        exercise.injected_questions.each_with_index do |question, index|
+          question.set(:'data-type', 'alphabetical-question-multipart')
+          question.add_class('alphabetical-multipart')
+          problem_letter = "(#{alphabet[index]})"
+          # I HAVE TO CHANGE THE DATA TYPE OF QUEStiON STEMS???
+
+          question.prepend(child:
+            <<~HTML
+              <div class="os-problem-container">
+                <a class='problem-letter' #{"href=##{question.id}-solution" if question.id.present?}>#{problem_letter}</a>
+                <span class='os-divider'> </span>
+                #{question.stimulus&.cut&.paste}
+                #{question.stem&.cut&.paste}
+              </div>
+            HTML
+          )
+
+          solution = question.solution
+          solution_id = "#{question.id}-solution"
+          solution&.replace_children(with:
+            <<~HTML
+              <a class='problem-letter' href='##{solution_id}'>#{problem_letter}</a>
+              <span class='os-divider'> </span>
+              <div class="os-solution-container">#{question.solution&.children}</div>
+            HTML
+          )
+          solution&.set(:'data-type', 'solution-part')
+          solution&.cut(to: solutions_clipboard)
+          question&.cut(to: questions_clipboard)
+        end
+
+        stimulus.set(:id, wrapper_id)
+        stimulus.set(:'data-type', 'exercise-question')
+
+        stimulus&.replace_children(with:
```

Comment:
I like this approach, but the code becomes confusing to read when things are renamed in place. Instead of using the stimulus as an outer wrapper (therefore having a variable _named_ stimulus that actually _represents_ the exercise-question) I would support creating a new wrapper for exercise-question, appending it to the injected-exercise (after cutting all of the initial contents aside from any exercise-level data) and pasting everything else inside the exercise-question wrapper. This would make it more clear what's going on here.